src/useGetSet/useGetSet.ts

```diff
@@ -0,0 +1,19 @@
+import { useRef, useCallback } from 'react';
+import { IInitialState, INextState, resolveHookState } from '../util/resolveHookState';
+import { useRerender } from '../index';
+
+export function useGetSet<S>(
+  initialState: IInitialState<S>
+): [get: () => S, set: (nextState: INextState<S>) => void] {
+  const rerender = useRerender();
+  const ref = useRef<S>(resolveHookState(initialState));
+  const get = useCallback(() => ref.current, []);
+  const set = useCallback(
+    (nextState: INextState<S>) => {
+      ref.current = resolveHookState(nextState, ref.current);
+      rerender();
+    },
+    [rerender]
+  );
+  return [get, set];
```

Comment:
Why not to use `useSafeState`, `useSyncedRef` and `useCallback`?

Besides this code will work - id prefer `native` state holder

Reply:
Sorry, I don't understand here.


> id prefer `native` state holder

what is it means here

Reply:
either useState nor useSafeState update the value immediately. That why this hook is need and `useRerender` is used in setter to control UI updates.

How `useSyncedRef` can be used here

Reply:
This implementation breaks `state flow` of react, and will break in strict mode, 4ex.

What i propose is to implement hook like this:

```ts
export function useGetSet<S>(
  initialState: IInitialState<S>
): [get: () => S, set: Dispatch<SetStateAction<S>>] {
  const [state, setState] = useSafeState(initialState);
  const stateRef = useSyncedRef(state);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  return [useCallback(() => stateRef.current, []), setState];
}
```

Reply:
This implementation doesn't return latest value in getter. /

Reply:
Well it shouldn't  - hook description says that it is like `useState`, but there is not a word about differenet behavior.

Therefore i assume that it should work exactly like `useState`, but return state getter instead of state itself.
I dont see any reason to deviate from react's state commitment flow. Or i simply dont know usecase for your implementation.

Reply:
make sense to me

Reply:
But how to write test, since I don't know when will react state change

Reply:
Just wrap all `set` calls with `act` method of testing library.

Reply:
Also, PR lack of readme changes.

Reply:
> Just wrap all `set` calls with `act` method of testing library.

```
  it('should set new value', () => {
    const { result } = setUp('foo');
    const [get, set] = result.current;
    act(() => set('bar'));
    const currentValue = get();
    expect(currentValue).toBe('bar');
  });
```
This test cannot pass /